docs/examples/basic-auth-api.md

```diff
@@ -0,0 +1,147 @@
+---
+title: "API using basic authentication"
+subtitle: "Example application that exposes an API behind basic authentication"
+type: 
+  - "pattern"
+platform:
+  - "awscdk"
+  - "tf-aws"
+  - "sim"
+language:
+  - "wing"
+githubURL: "https://github.com/winglang/examples.git && cd examples/examples/api-basic-auth"
+coverImage: "/img/examples/basic-api-auth.png"
+coverImageInPage: true
+resources:
+  - label: "Understand preflight and inflight concepts"
+    href: "/docs/concepts/inflights"
+  - label: "Explore the Wing api resource"
+    href: "/docs/api/standard-library/cloud/api"
+authors:
+  - name: "David Boyne"
+    role: "Developer Advocate, Wing"
+    twitter: "https://twitter.com/boyney123"
+    github: "https://github.com/boyney123"
+cloudResources:
+  - api
+---
+
+This example application deploys an [API](/docs/api/standard-library/cloud/api) with a GET route called `/user`. The route is behind basic authentication and returns a 401 if un-authorized.
+
+In this example a custom **BasicAuth** class is created. This class exposes a public inflight method (called verify) that is used within the API to verify the request. 
+
+> **Did you know resources have a preflight and inflight API?** [Preflight](/docs/concepts/inflights) code runs once, at compile time whereas [inflight](/docs/concepts/inflights) code is run at runtime. In this example we create the BasicAuth and API resources on preflight (default for Wing files) and then use the BasicAuth method (verify) in the API itself (on inflight). To learn more you can read [preflight and inflight](/docs/concepts/inflights#preflight-code) core concepts.
+
+
+```js example playground
```

Comment:
It will be cool to refresh the example once we have some dedicated middleware support for cloud.Api (https://github.com/winglang/wing/issues/3518)

Reply:
Yeah sounds good to me. Any idea when that would be?